fuel-block-producer/src/block_producer.rs

```diff
@@ -144,24 +143,19 @@ impl Trait for Producer {
             Some(height) => height,
         } + 1u64.into();
         let checked = if self.config.utxo_validation {
-            CheckedTransaction::check(
-                transaction,
-                height.into(),
-                &self.config.consensus_params,
-            )?
+            let (transaction, _) = transaction
+                .into_checked(height.into(), &self.config.consensus_params)?
+                .into();
```

Comment:
Not sure how others feel, but for me this is a bit unclear, I have no idea what does `into()` return here by just looking at the code

Reply:
It is cast back to `Transaction`.  Actually, we can remove this check here, because it is checked during execution inside. 

<img width="898" alt="image" src="https://user-images.githubusercontent.com/18346821/197865003-3ffe82ff-a43e-45a6-aaf9-b415d620321f.png">

I guess we do this check here because in the future, we want to remove checking it from the executor. Am I right? Or we can remove it?=D @Voxelot

Reply:
Ah yeah this seems redundant.